src/analysis/score.rs

```diff
@@ -24,6 +30,94 @@ pub fn row_accuracy(outputs: &Matrix<f64>, targets: &Matrix<f64>) -> f64 {
     accuracy(outputs.iter_rows(), targets.iter_rows())
 }
 
+/// Returns the precision score for 2 class classification.
+/// true-positive / (true-positive + false-positive)
+pub fn precision<'a, I, T>(outputs: I, targets: I) -> f64
+    where I: ExactSizeIterator<Item=&'a T>,
+          T: 'a + PartialEq + Zero + One
+{
+    assert!(outputs.len() == targets.len());
```

Comment:
I agree that panicking is the best course of action, but it may be worth adding a custom error message here. Maybe, "Output and target iterators must have the same length."